plugins/workspace-plugin/src/ca-cert.ts

```diff
@@ -0,0 +1,53 @@
+/**********************************************************************
+ * Copyright (c) 2019-2020 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ ***********************************************************************/
+
+import * as fs from 'fs';
+import * as path from 'path';
+
+const PUBLIC_CRT_PATH = '/public-certs';
+const SS_CRT_PATH = '/tmp/che/secret/ca.crt';
+
+const CA_BUNDLE_PATH = '/tmp/ca-bundle.crt';
+
+let bundle: string | undefined = undefined;
+
+export async function getAuthorityCertificate(): Promise<string | undefined> {
+  if (bundle !== undefined) {
+    return bundle;
+  }
+
+  const certificateAuthority: Buffer[] = [];
+
+  if (fs.existsSync(SS_CRT_PATH)) {
+    certificateAuthority.push(fs.readFileSync(SS_CRT_PATH));
```

Comment:
await readFile(path, 'utf-8')

Reply:
All overloads for readFile return void, not a promise. Having await here will not work

For the encoding documentation says:
> If the encoding option is specified then this function returns a string. Otherwise it returns a buffer.

We operate with Buffer.

Reply:
Ok, probably I read about 'fs', not 'fs-extra'

Reply:
if it's buffer then only await readFile(path); but it doesn't return void

are you looking doc of fs-extra ? or the old/default fs that use callbacks ?

Reply:
there are several examples in che-theia: https://github.com/eclipse/che-theia/blob/4ea8a5b287f0605a721d1b0d07bcef572c04ee1a/extensions/eclipse-che-theia-user-preferences/src/node/che-theia-preferences-synchronizer.ts#L53

Reply:
> are you looking doc of fs-extra ? or the old/default fs that use callbacks ?

I have already reworked it, thanks.